apps/files/src/components/Collaborators/CollaboratorsEditOptions.vue

```diff
@@ -51,19 +72,35 @@ export default {
       required: false
     },
     expirationDate: {
-      type: String,
+      type: Date,
       required: false
+    },
+    existingCollaboratorType: {
+      type: [Object, String],
+      required: false,
+      validator: function (value) {
+        return ['user', 'group'].indexOf(value) > -1 || value === null
+      }
     }
   },
+
   data () {
     return {
       selectedRole: null,
-      additionalPermissions: null
+      additionalPermissions: null,
+      enteredExpirationDate: null
     }
   },
+
   computed: {
-    $_ocCollaborators_expirationSupported () {
-      return false
+    ...mapGetters(['capabilities']),
+
+    editingUser () {
```

Comment:
would be nice to make all these private by prefixing with `$_`.

the other day we did a test and noticed that such public members are accessible from the outside when accessing the vue object instance

Reply:
Then something must be wrong if the methods were accessible from the Vue instance since this is not how Vue works. If you want to have methods like that they need to be directly assigned to the global instance. Otherwise, you need to do a few gymnastics to get to them (e.g. via `$parent$)

And if we make private everything then we are limiting ourselves and also introducing another step that outside developers would need to understand when contributing.

^ just my opinion

Reply:
I think the way to access is by using refs. You could then call methods on child vues.

I would only make stuff private if it's not intended to be used outside like in parent vues.

Reply:
here's an example: https://github.com/owncloud/phoenix/blob/master/apps/files/src/components/Collaborators/NewCollaborator.vue#L123

we deliberately exposed the "focus()" method on the autocomplete component from ODS.

Reply:
Yeah, but that a bit different since this is coming from a design system component where indeed it does make sense for defining the private methods specifically. But in other cases, it is kind of an anti-pattern to access it like this since with that we are making the components dependent on each other. That's where the store should come into play.

Reply:
Another advantage of marking private members is that you directly see whether you're accessing a local private member vs one imported from the store when looking at the markup.

I'm fine solidifying either decision as long as we are consistent.

cc @kulmann as we discussed the public member thing before

Reply:
From Vue docs:

> **Use module scoping to keep private functions inaccessible from the outside**. If that’s not possible, always use the $_ prefix for custom private properties in a plugin, mixin, etc that should not be considered public API. Then to avoid conflicts with code by other authors, also include a named scope (e.g. $_yourPluginName_).

Reply:
ah, so those private guidelines are for plugins and mixins, not Vue components ?

I'm fine removing the prefixes then

Reply:
> ah, so those private guidelines are for plugins and mixins, not Vue components ?

Yes.